src/Data/Fin/Relation/Unary/Top.agda

```diff
@@ -0,0 +1,84 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The 'top' view of Fin
+--
+-- This is an example of a view of (elements of) a datatype,
+-- here i : Fin (suc n), which exhibits every such i as
+-- * either, i = fromℕ n
+-- * or, i = inject₁ j for a unique j : Fin n
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Fin.Relation.Unary.Top where
+
+open import Data.Nat.Base using (ℕ; zero; suc; _<_)
+open import Data.Fin.Base using (Fin; zero; suc; toℕ; fromℕ; inject₁)
+open import Data.Fin.Properties using (toℕ-fromℕ; toℕ-inject₁; inject₁ℕ<)
+open import Relation.Binary.PropositionalEquality.Core
+open import Relation.Nullary.Negation using (contradiction)
+
+private
+  variable
+    n : ℕ
+    i : Fin (suc n)
+    j : Fin n
+
+------------------------------------------------------------------------
+-- The View, considered as a unary relation on Fin (suc n)
+
+-- NB `Data.Fin.Properties.fromℕ≢inject₁` establishes that the following
+-- inductively defined family on `Fin (suc n)` has constructors which
+-- target *disjoint* instances of the family; and hence the interpretations
+-- of the View constructors will also be disjoint
+
+data View : ∀ {n} (j : Fin n) → Set where
+  ‵fromℕ : View (fromℕ n)
+  ‵inj₁ : {j : Fin n} → View j → View (inject₁ j)
+
+pattern ‵inject₁ {n} j = ‵inj₁ {n = n} {j = j} _
+
+-- The view covering function, witnessing soundness of the view
+
+view-zero : ∀ n → View {suc n} zero
+view-zero zero    = ‵fromℕ
+view-zero (suc n) = ‵inj₁ (view-zero n)
+
+view : (j : Fin n) → View j
+view zero = view-zero _
+view (suc i) with view i
+... | ‵fromℕ  = ‵fromℕ
+... | ‵inject₁ j = ‵inj₁ (view (suc j))
+
+-- Interpretation of the view constructors
+
+⟦_⟧ : View {n} j → Fin n
+⟦ ‵fromℕ ⟧     = fromℕ _
+⟦ ‵inject₁ j ⟧ = inject₁ j
+
+-- Completeness of the view
+
+view-complete : (v : View {n} j) → ⟦ v ⟧ ≡ j
```

Comment:
Here, `{n}` may even be removed? Will investigate... and if so, does/would/might that assuage your concern(s) above?